qiskit/providers/aer/jobs/aerjobset.py

```diff
@@ -218,6 +232,63 @@ def _get_worker_result(self, worker: int, timeout: Optional[float] = None):
                     "Timeout while waiting for JobSet results")
         return result
 
+    def _merge_results(self, result_list: List[Result]):
+        if len(result_list) == 1:
+            return result_list[0]
+
+        master_result = result_list[0].to_dict()
+        sub_result = result_list[1].to_dict()
+        result_list = []
+
+        for (_master_result, _sub_result) in zip(master_result["results"], sub_result["results"]):
+            result_list.append(self._merge_exp(_master_result, _sub_result))
+        master_result["results"] = result_list
+        return Result.from_dict(master_result)
```

Comment:
If you are modifying an existing master result, you should just be updating it, not constructing a new one I think?

Reply:
Ok. I changed the code to just return master result.